### PR TITLE
NoBlankLinesAfterPhpdoc - Add T_NAMESPACE in array of forbidden successors

### LINKED ISSUE #5111: 'header_comment' contradicts with 'no_blank_lines_after_phpdoc' and 'single_blank_line_before_namespace'
## Bug report

PHP 7.4.9, php-cs-fixer v2.16.4

When setting the `separate` option of `header_comment` to `both`, and configuring `no_blank_lines_after_phpdoc` and `single_blank_line_before_namespace` to both `true`, the latter two shows up when after `header_comment` did its job.

### Code snippet that reproduces the problem
1. First run, create a class with no header.
```php
<?php

namespace Foo;

class Bar
{
}
```
After the first run, the error show will be `header_comment`, which is expected.

2. Next run, create additions that will violate any other rules, e.g. `native_constant_invocation`.
```php
<?php

/**
 * An arbitrary file header
 */

namespace Foo;

class Bar
{
    public $separator = DIRECTORY_SEPARATOR;
}
```
After this run, errors shown would be `native_constant_invocation`, `no_blank_lines_after_phpdoc`, and `single_blank_line_before_namespace`.

### PR DESCRIPTION
**Pull request by paulbalandan**
Fixes #5111 

When used, `no_blank_lines_after_phpdoc` removes the whitespace between the file-level doc-block and the namespace declaration. This should not be the case as namespaces are not considered structural elements that can be documented with doc-blocks. Besides, the file-level doc-block belongs to the file and NOT on the namespace.

See more: https://docs.phpdoc.org/3.0/guide/getting-started/what-is-a-docblock.html

### CODE REVIEW

**Comment by SpacePossum** [comment]
Hi and thanks for your PR!

Looking good already. Could you adjust the test to only test the single issue here, maybe like:

```php
    public function testWhitespaceInDocBlockAboveNamespaceIsNotTouched()
    {
        $expected = '<?php
/**
 * This is a file-level docblock.
 */

namespace Foo\Bar\Baz;
';
        $this->doTest($expected);
    }
```

Thanks!

File: src/Fixer/Phpdoc/NoBlankLinesAfterPhpdocFixer.php [hunk 1]
```diff
@@ -81,6 +81,7 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
             T_BREAK,
             T_DECLARE,
             T_USE,
+            T_NAMESPACE,
```
Comment (SpacePossum) [inline src/Fixer/Phpdoc/NoBlankLinesAfterPhpdocFixer.php:84]: please sort this list alfabetical

Comment (paulbalandan) [reply src/Fixer/Phpdoc/NoBlankLinesAfterPhpdocFixer.php:80]: Done.

**Review by SpacePossum — APPROVED** [review]

**Comment by keradus** [comment]
@paulbalandan CI is failing for proposed changes

**Comment by paulbalandan** [comment]
@keradus or @SpacePossum , is it an acceptable fix to just remove the integration of the fixer against `single_blank_line_after_namespace`? Before this PR, `no_blank_lines_after_phpdoc` considers `T_NAMESPACE` as documentable that's why it gets an integration/priority check with `single_blank_line_after_namespace`. Now that this skips `T_NAMESPACE`, I think the integration/priority checks should be dropped too.

**Comment by keradus** [comment]
if there is no longer priority conflict between to rules, move test from `Integration/priority/` to `Integration/misc` and drop `tests/AutoReview/FixerFactoryTest.php:143:            [$fixers['no_blank_lines_after_phpdoc'], $fixers['single_blank_line_before_namespace']]` row

**Comment by paulbalandan** [comment]
I have fixed the integration failure. Not sure if the removal of the priority comment in `SingleBlankLineBeforeNamespace` is warranted.

**Comment by keradus** [comment]
please avoid squashing the commits in future. now, I have no clue what are the newest changes and have to review from scratch :(

**Comment by keradus** [comment]
> Not sure if the removal of the priority comment (...)

we have tests for it. if CI is happy, all good

**Review by kubawerlos — APPROVED** [review]

**Comment by keradus** [comment]
Thank you @paulbalandan.

**Pull request merged**